x-pack/test/functional/services/ml/cases.ts

```diff
@@ -38,19 +41,23 @@ export function MachineLearningCasesProvider(
       await cases.casesTable.goToFirstListedCase();
     },
 
+    async assertBasicCaseProps(params: CaseParams) {
+      await this.openCaseInCasesApp(params.tag);
+      await elasticChart.setNewChartUiDebugFlag(true);
```

Comment:
I think this row isn't needed in this particular service method, so I'd suggest to move it back to `assertCaseWithAnomalySwimLaneAttachment` and `assertCaseWithAnomalyChartsAttachment` and also make sure that we disable the debug flag again after assertions are done (or we leave it away from the service methods completely and let the test file handle it, where we already call it).

Reply:
Unfortunately, it isn't not going to work. We need to set the debug flag _**before**_ the chart is rendered. And when we perform the navigation here, it involves the page refresh, hence I'm also not able to set this flag before because we will lose the state on reload. Alternatively, I can pass a callback to this method, which will to be invoked right after hitting the URL. What do you think?

Reply:
I like the idea with the callback :+1: